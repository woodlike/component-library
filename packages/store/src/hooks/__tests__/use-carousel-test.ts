import { renderHook, act } from '@testing-library/react-hooks';
import { previous, next, useCarousel, jump } from '..';

describe('UseCarousel', () => {
  describe('useCarousel()', () => {
    it('should return a default coordinate state of 0', () => {
      const { result } = renderHook(() => useCarousel(4));
      const { coordinate } = result.current;
      expect(coordinate).toBe(0);
    });

    it('should return the previous state coordinate if the current is outsite the length scope', () => {
      const { result } = renderHook(() => useCarousel(4));
      act(() => {
        result.current.jump(2);
      });

      expect(result.current.coordinate).toBe(-50);

      act(() => {
        result.current.jump(5);
      });

      expect(result.current.coordinate).toBe(-50);
    });

    it.each([0, 1, 2, 3, 4, 5, 6, 7, 8])(
      'should return the coordinate of %d',
      current => {
        const { result } = renderHook(() => useCarousel(9));

        act(() => {
          result.current.jump(current);
        });

        expect(result.current.coordinate).toMatchSnapshot();
      },
    );

    it.each([8, 7, 6, 5, 4, 3, 2, 1, 0])(
      'should return the previous coordinate of %d',
      current => {
        const { result } = renderHook(() => useCarousel(9));

        act(() => {
          result.current.jump(current);
          result.current.previous();
        });

        expect(result.current.coordinate).toMatchSnapshot();
      },
    );

    it.each([0, 1, 2, 3, 4, 5, 6, 7, 8])(
      'should return the the next coordinate of %d',
      current => {
        const { result } = renderHook(() => useCarousel(9));

        act(() => {
          result.current.jump(current);
          result.current.next();
        });

        expect(result.current.coordinate).toMatchSnapshot();
      },
    );
  });

  describe('jump()', () => {
    it.each([0, 1, 2, 3, 4])('should jump to %d', current => {
      expect(jump(current, 5)).toMatchSnapshot();
    });
  });

  describe('previous()', () => {
    it.each([10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0])(
      'should return the previous coordinate of %d',
      current => {
        expect(previous(current, 11)).toMatchSnapshot();
      },
    );
  });

  describe('next()', () => {
    it.each([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10])(
      'should return the next coordinate of %d',
      current => {
        expect(next(current, 11)).toMatchSnapshot();
      },
    );
  });
});
