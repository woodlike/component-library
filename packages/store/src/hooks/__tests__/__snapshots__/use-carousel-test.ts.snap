// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UseCarousel jump() should jump to 0 1`] = `-0`;

exports[`UseCarousel jump() should jump to 1 1`] = `-20`;

exports[`UseCarousel jump() should jump to 2 1`] = `-40`;

exports[`UseCarousel jump() should jump to 3 1`] = `-60`;

exports[`UseCarousel jump() should jump to 4 1`] = `-80`;

exports[`UseCarousel move() should return the percentage and coordinate when the clientX is 200 on each item 0 1`] = `
Object {
  "coordinate": 0.555,
  "percentage": -3.333,
}
`;

exports[`UseCarousel move() should return the percentage and coordinate when the clientX is 200 on each item 1 1`] = `
Object {
  "coordinate": 1.111,
  "percentage": -3.333,
}
`;

exports[`UseCarousel move() should return the percentage and coordinate when the clientX is 200 on each item 2 1`] = `
Object {
  "coordinate": 1.667,
  "percentage": -3.333,
}
`;

exports[`UseCarousel move() should return the percentage and coordinate when the clientX is 200 on each item 3 1`] = `
Object {
  "coordinate": 2.222,
  "percentage": -3.333,
}
`;

exports[`UseCarousel move() should return the percentage and coordinate when the clientX is 200 on each item 4 1`] = `
Object {
  "coordinate": 2.778,
  "percentage": -3.333,
}
`;

exports[`UseCarousel move() should return the percentage and coordinate when the clientX is 200 on each item 5 1`] = `
Object {
  "coordinate": 3.333,
  "percentage": -3.333,
}
`;

exports[`UseCarousel next() should return the next coordinate of 0 1`] = `-9.090909090909092`;

exports[`UseCarousel next() should return the next coordinate of 1 1`] = `-18.181818181818183`;

exports[`UseCarousel next() should return the next coordinate of 2 1`] = `-27.272727272727273`;

exports[`UseCarousel next() should return the next coordinate of 3 1`] = `-36.36363636363637`;

exports[`UseCarousel next() should return the next coordinate of 4 1`] = `-45.45454545454545`;

exports[`UseCarousel next() should return the next coordinate of 5 1`] = `-54.54545454545455`;

exports[`UseCarousel next() should return the next coordinate of 6 1`] = `-63.63636363636363`;

exports[`UseCarousel next() should return the next coordinate of 7 1`] = `-72.72727272727273`;

exports[`UseCarousel next() should return the next coordinate of 8 1`] = `-81.81818181818181`;

exports[`UseCarousel next() should return the next coordinate of 9 1`] = `-90.9090909090909`;

exports[`UseCarousel next() should return the next coordinate of 10 1`] = `-0`;

exports[`UseCarousel previous() should return the previous coordinate of 0 1`] = `-90.9090909090909`;

exports[`UseCarousel previous() should return the previous coordinate of 1 1`] = `-0`;

exports[`UseCarousel previous() should return the previous coordinate of 2 1`] = `-9.090909090909092`;

exports[`UseCarousel previous() should return the previous coordinate of 3 1`] = `-18.181818181818183`;

exports[`UseCarousel previous() should return the previous coordinate of 4 1`] = `-27.272727272727273`;

exports[`UseCarousel previous() should return the previous coordinate of 5 1`] = `-36.36363636363637`;

exports[`UseCarousel previous() should return the previous coordinate of 6 1`] = `-45.45454545454545`;

exports[`UseCarousel previous() should return the previous coordinate of 7 1`] = `-54.54545454545455`;

exports[`UseCarousel previous() should return the previous coordinate of 8 1`] = `-63.63636363636363`;

exports[`UseCarousel previous() should return the previous coordinate of 9 1`] = `-72.72727272727273`;

exports[`UseCarousel previous() should return the previous coordinate of 10 1`] = `-81.81818181818181`;

exports[`UseCarousel useCarousel() should return the coordinate of 0 1`] = `-0`;

exports[`UseCarousel useCarousel() should return the coordinate of 1 1`] = `-11.11111111111111`;

exports[`UseCarousel useCarousel() should return the coordinate of 2 1`] = `-22.22222222222222`;

exports[`UseCarousel useCarousel() should return the coordinate of 3 1`] = `-33.333333333333336`;

exports[`UseCarousel useCarousel() should return the coordinate of 4 1`] = `-44.44444444444444`;

exports[`UseCarousel useCarousel() should return the coordinate of 5 1`] = `-55.55555555555556`;

exports[`UseCarousel useCarousel() should return the coordinate of 6 1`] = `-66.66666666666667`;

exports[`UseCarousel useCarousel() should return the coordinate of 7 1`] = `-77.77777777777777`;

exports[`UseCarousel useCarousel() should return the coordinate of 8 1`] = `-88.88888888888889`;

exports[`UseCarousel useCarousel() should return the previous coordinate of 0 1`] = `-88.88888888888889`;

exports[`UseCarousel useCarousel() should return the previous coordinate of 1 1`] = `-0`;

exports[`UseCarousel useCarousel() should return the previous coordinate of 2 1`] = `-11.11111111111111`;

exports[`UseCarousel useCarousel() should return the previous coordinate of 3 1`] = `-22.22222222222222`;

exports[`UseCarousel useCarousel() should return the previous coordinate of 4 1`] = `-33.333333333333336`;

exports[`UseCarousel useCarousel() should return the previous coordinate of 5 1`] = `-44.44444444444444`;

exports[`UseCarousel useCarousel() should return the previous coordinate of 6 1`] = `-55.55555555555556`;

exports[`UseCarousel useCarousel() should return the previous coordinate of 7 1`] = `-66.66666666666667`;

exports[`UseCarousel useCarousel() should return the previous coordinate of 8 1`] = `-77.77777777777777`;

exports[`UseCarousel useCarousel() should return the the next coordinate of 0 1`] = `-11.11111111111111`;

exports[`UseCarousel useCarousel() should return the the next coordinate of 1 1`] = `-22.22222222222222`;

exports[`UseCarousel useCarousel() should return the the next coordinate of 2 1`] = `-33.333333333333336`;

exports[`UseCarousel useCarousel() should return the the next coordinate of 3 1`] = `-44.44444444444444`;

exports[`UseCarousel useCarousel() should return the the next coordinate of 4 1`] = `-55.55555555555556`;

exports[`UseCarousel useCarousel() should return the the next coordinate of 5 1`] = `-66.66666666666667`;

exports[`UseCarousel useCarousel() should return the the next coordinate of 6 1`] = `-77.77777777777777`;

exports[`UseCarousel useCarousel() should return the the next coordinate of 7 1`] = `-88.88888888888889`;

exports[`UseCarousel useCarousel() should return the the next coordinate of 8 1`] = `-0`;
